File: apps/workshop-java/workshop/src/main/java/com/theagilemonkeys/workshop/controllers/SearchController.java
package com.theagilemonkeys.workshop.controllers;

import com.theagilemonkeys.workshop.services.SemanticSearchService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.ResponseBody;

import com.theagilemonkeys.ellmental.semanticsearch.SearchOutput;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.concurrent.CompletableFuture;
@Controller
public class SearchController {
    private final SemanticSearchService semanticSearchService;

    @Autowired
    public SearchController(SemanticSearchService semanticSearchService) {
        this.semanticSearchService = semanticSearchService;
    }

    @GetMapping("/search")
    @ResponseBody
    public CompletableFuture<SearchOutput> search(@RequestParam String query) {
        // implement the search functionality using the semanticSearchService
    }
}

File: apps/workshop-kotlin/workshop/src/main/kotlin/com/theagilemonkeys/workshop/controllers/SearchController.kt
package com.theagilemonkeys.workshop.controllers

import com.theagilemonkeys.ellmental.semanticsearch.SearchOutput
import com.theagilemonkeys.workshop.services.SemanticSearchService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam


@Controller
class SearchController(@Autowired private val semanticSearchService: SemanticSearchService) {
    @GetMapping("/search")
    suspend fun search(@RequestParam query: String): SearchOutput {
        return semanticSearchService.search(query)
    }
}